internal/pkg/v2/ui/common/routable.go

```diff
@@ -0,0 +1,27 @@
+/*******************************************************************************
+ * Copyright 2020 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package common
```

Comment:
Why `ui` rather than `api` in the package path?

Reply:
It's a generic user-interface-level concern.

api is under http -- which is transport-specific.

It's placed outside of http in a common package because it is intended to be reused by future alternative transport implementations.

Reply:
When i see `ui`, I have the expectation that there will be visual user interface code.... I think this will be confusing. Would `usecase` be more descriptive?

Reply:
No, it would not.  Use-case is already in use; it's an application-level concept.

FWIW, I'm using the layers common to a domain-driven design application that uses traditional Layers Architecture [Buschmann, Frank, et al. 1996.  _Pattern-Oriented Software Architecture, Volume 1: A System of Patterns_. New York: Wiley.]. 

The user interface layer is simply the top-most layer that interacts with the external world; i.e. the user.  It's an extremely applicable label for the responsibilities assigned to this layer in this implementation.

It's unfortunate that user interface equates conceptually to a screen with windows and dialogs.  The term is actually quite generic:

![image](https://user-images.githubusercontent.com/464327/73568474-146b9280-442e-11ea-9041-8e529fc3925e.png)

Reply:
Ok, you have sold me... :-)